File: tests/dat/actions/hello.js
// Licensed to the Apache Software Foundation (ASF) under one or more contributor
// license agreements; and to You under the Apache License, Version 2.0.

/**
 * Hello, world.
 */
function main(params) {
    var greeting = 'hello, ' + params.payload + '!'
    console.log(greeting);
    return {payload: greeting}
}

File: tests/dat/actions/hello.swift
// Licensed to the Apache Software Foundation (ASF) under one or more contributor
// license agreements; and to You under the Apache License, Version 2.0.

/**
 * Hello world as a Swift Whisk action.
 */
func main(args: [String:Any]) -> [String:Any] {
    if let name = args["name"] as? String {
        return [ "greeting" : "Hello \(name)!" ]
    } else {
        return [ "greeting" : "Hello stranger!" ]
    }
}
